cmd/run.go

```
@@ -278,12 +288,74 @@ func runMigrations(c db.Config, name string) {
+func newEtherman(c config.Config, st *state.State) (*etherman.Client, error) {
+	ethman, err := etherman.NewClient(c.Etherman, c.NetworkConfig.L1Config, nil)
 	if err != nil {
 		return nil, err
 	}
-	return etherman, nil
+	da, err := newDataAvailability(c, st, ethman, false)
+	if err != nil {
+		return nil, err
+	}
+	return etherman.NewClient(c.Etherman, c.NetworkConfig.L1Config, da)
+}
+
+func newDataAvailability(c config.Config, st *state.State, etherman *etherman.Client, isSequenceSender bool) (*dataavailability.DataAvailability, error) {
+	var trustedSequencerURL string
+	var err error
+	if !c.IsTrustedSequencer {
+		if c.Synchronizer.TrustedSequencerURL != "" {
+			trustedSequencerURL = c.Synchronizer.TrustedSequencerURL
+		} else {
+			log.Debug("getting trusted sequencer URL from smc")
+			trustedSequencerURL, err = etherman.GetTrustedSequencerURL()
+			if err != nil {
+				log.Fatal("error getting trusted sequencer URI. Error: %v", err)
+			}
+		}
+		log.Debug("trustedSequencerURL ", trustedSequencerURL)
+	}
+	zkEVMClient := client.NewClient(trustedSequencerURL)
+
+	// Backend specific config
+	daProtocolName, err := etherman.GetDAProtocolName()
+	if err != nil {
+		log.Fatal("error getting data availability protocol name: %v", err)
+	}
+	var daBackend dataavailability.DABackender
+	switch daProtocolName {
+	case string(dataavailability.DataAvailabilityCommittee):
+		var pk *ecdsa.PrivateKey
+		var err error
+		if isSequenceSender {
+			_, pk, err = etherman.LoadAuthFromKeyStore(c.SequenceSender.PrivateKey.Path, c.SequenceSender.PrivateKey.Password)
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+		dacAddr, err := etherman.GetDAProtocolAddr()
+		if err != nil {
+			log.Fatal("error getting trusted sequencer URI. Error: %v", err)
+		}
+
+		daBackend, err = datacommittee.New(
+			c.Etherman.URL,
+			dacAddr,
+			c.SequenceSender.L2Coinbase,
+			pk,
+			&dataCommitteeClient.Factory{},
+		)
+		if err != nil {
+			return nil, err
+		}
```

Shall we add default statement and return an error perhaps in order to fail fast, because it would mean we have retrieved some unknown DA protocol name?